core/lib/makeswift/components/customer-group-slot/customer-group-slot.makeswift.tsx

```diff
@@ -0,0 +1,141 @@
+import { Checkbox, Combobox, Group, List, Slot, Style } from '@makeswift/runtime/controls';
+import { ReactNode } from 'react';
+import useSWR from 'swr';
+import { z } from 'zod';
+
+import { CustomerGroupsSchema, CustomerGroupsType } from '~/app/api/customer/groups/route';
+import { GetCustomerGroupResponse } from '~/client/queries/get-customer';
+import { runtime } from '~/lib/makeswift/runtime';
+
+const CustomerGroupSchema = z.object({
+  customer: z.object({
+    customerGroupId: z.number(),
+  }),
+});
+
+interface Props {
+  className: string;
+  slots?: Array<{ group?: string; slot: ReactNode }>;
+  simulateGroup: boolean;
+  simulatedGroup?: string;
+  noGroupSlot: ReactNode;
+}
+
+async function getAllCustomerGroups(): Promise<CustomerGroupsType> {
+  const response = await fetch('/api/customer/groups');
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const data: unknown = await response.json();
+  const groups = CustomerGroupsSchema.parse(data);
+
+  return groups;
+}
+
+async function fetchCustomerGroupData(): Promise<GetCustomerGroupResponse> {
+  const response = await fetch('/api/customer/group');
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const data: unknown = await response.json();
+  const group = CustomerGroupSchema.parse(data);
+
+  return group;
+}
+
+function CustomerGroupSlot({
+  className,
+  slots,
+  simulateGroup,
+  simulatedGroup = 'no-group',
+  noGroupSlot,
+}: Props) {
+  const allSlots = slots?.concat({ group: 'no-group', slot: noGroupSlot });
+
+  const { data, isLoading } = useSWR<GetCustomerGroupResponse>(
+    '/api/customer/group',
+    fetchCustomerGroupData,
+  );
+
+  // If we're simulating the group, render the selected group slot, otherwise use the group from the API. Coalesce to noGroupSlot if no group is found
+  const groupSlot = simulateGroup
+    ? (allSlots?.find((s) => s.group === simulatedGroup)?.slot ?? (
+        // If the simulatedGroup does not exist in the list of slots
+        <div className="p-4 text-center text-lg text-gray-400">
+          This group needs to be added to "Targeted customer groups".
+        </div>
+      ))
+    : (allSlots?.find((s) => s.group === `${data?.customer?.customerGroupId}`)?.slot ?? (
+        <div className="p-4 text-center text-lg text-gray-400">
+          This group needs to be added to "Targeted customer groups".
+        </div>
+      ));
```

Comment:
Can we have some intermediate values to make this a bit easier to reason about? In particular, can we extract the slots we're finding?